src/sentry/models/project.py

```diff
@@ -313,6 +305,21 @@ def transfer_to(self, team):
                 team=team,
             )
 
+        # Both environments and releases are bound at an organization level.
+        # Due to this, when you transfer a project into another org, we have to
+        # handle this behavior somehow. We really only have two options here:
+        # * Copy over all releases/environments into the new org and handle de-duping
```

Comment:
I think this is an objective improvement over the current state of things, but we should look at moving towards this path (copying environments at least, haven't put a lot of thought into releases specifically, but I assume you could make a similar argument there) as soon as possible.